### PR TITLE
7zip: Build 7z.dll

### PR DESCRIPTION
**Pull request by Biswa96**
Fixes https://github.com/msys2/MINGW-packages/issues/14637

### CODE REVIEW

**Comment by Biswa96** [comment]
@dscho Would you like to review this pull request please?

File: mingw-w64-7zip/PKGBUILD [hunk 1]
```diff
@@ -30,13 +30,15 @@ prepare () {
...
 
+  make -C CPP/7zip/Bundles/Format7zF -f makefile.gcc
   make -C CPP/7zip/UI/Console -f makefile.gcc
   make -C CPP/7zip/Bundles/SFXCon -f makefile.gcc
 }
 
 package() {
   cd "${srcdir}"
 
+  install -D -m755 CPP/7zip/Bundles/Format7zF/_o/7z.dll "${pkgdir}"${MINGW_PREFIX}/bin/7z.dll
```
Comment (dscho) [inline mingw-w64-7zip/PKGBUILD:41]: Is this the only `.dll` that now needs to be copied?

**Review by dscho — COMMENTED** [review]
Thank you for fixing this!

File: mingw-w64-7zip/PKGBUILD (hunk 1 above)
Comment (Biswa96) [reply mingw-w64-7zip/PKGBUILD:41]: That's the only DLL compiled in that directory.

**Comment by Biswa96** [comment]
I am not sure how to fix the arm64 build. May I ask for help please? @jeremyd2019

File: mingw-w64-7zip/PKGBUILD (hunk 1 above)
Comment (dscho) [reply mingw-w64-7zip/PKGBUILD:41]: I just wonder how future-proof it is to hard-code the path to the currently-only DLL file.

Comment (Biswa96) [reply mingw-w64-7zip/PKGBUILD:41]: I did not fully understand the question. Could you rephrase the question please? A diff suggestion may help.

BTW, I am not good at making things stable 👶

**Comment by jeremyd2019** [comment]
```
../../../../C/Sha1Opt.c:313:49: warning: call to undeclared function 'vsha1h_u32'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
```
looks familiar.  Probably needs to be patched to include some arm intrinsic header, to use a form of target attribute that clang accepts, and to `#define` something to convince the arm header to actually include the required intrinsics even though the instructions aren't enabled globally.

**Comment by jeremyd2019** [comment]
Ah, see the patch for Sha256Opt.c in https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-7zip/clang-arm-target-fixes.patch and try to apply similar to Sha1Opt.c?

File: mingw-w64-7zip/PKGBUILD (hunk 1 above)
Comment (dscho) [reply mingw-w64-7zip/PKGBUILD:41]:
```suggestion
  install -D -m755 $(find CPP/7zip/Bundles/ -name \*.dll) "${pkgdir}"${MINGW_PREFIX}/bin/
```

Comment (Biswa96) [reply mingw-w64-7zip/PKGBUILD:41]: It did not change anything in the package though. Should I add that for any other reason?

Comment (dscho) [reply mingw-w64-7zip/PKGBUILD:41]: > It did not change anything in the package though.

Of course not. That would be a bug.

> Should I add that for any other reason?

Certainly. The `find` command is robust against future changes in 7-Zip. Hard-coding file names is always fragile.

Comment (Biswa96) [reply mingw-w64-7zip/PKGBUILD:41]: > The find command is robust against future changes in 7-Zip.

According to that point, there is a possibility that Format7zF/_o/7z.dll path will be changed. Would not there a similar chance of having multiple DLL files in Format7zF/_o/ folder?

Comment (dscho) [reply mingw-w64-7zip/PKGBUILD:41]: Let's not jump from one extreme (totally fragile code) to the other extreme (totally over-engineered yet robust code).
